lib/core/i18n/i18n.js

```diff
@@ -16,28 +17,20 @@ function isSupported(locale) {
   return (supported_languages.indexOf(locale.substr(0, 2)) >= 0);
 }
 
-function setLocale(locale) {
-  i18n.setLocale(locale.substr(0, 2));
-}
-
-function setDefaultLocale() {
-  osLocale().then(setLocale).catch();
-}
-
 function setOrDetectLocale(locale) {
-  if (locale && !isSupported(locale)) {
-    console.log("===== locale " + locale + " not supported =====");
-  }
-  if (locale) {
-    return i18n.setLocale(locale.substr(0, 2));
+  const how = locale ? 'specified' : 'detected';
+  let _locale = locale || osLocale.sync();
+  _locale = _locale.substr(0, 2);
+  if (_locale && !isSupported(_locale)) {
+    console.log(`===== locale ${_locale} ${how} but not supported, default: en =====`);
```

Comment:
You could use console.warn and also put `.yellow` or other colour for the text so it pops up more.

Reply:
Done.